tests/testthat/test-foreign-key-constraints.R

```diff
@@ -108,3 +112,41 @@ test_that("empty key constraints returns appropriate messages", {
 
   expect_false(keys$is_serial("table", "id"))
 })
+
+test_that("can get network table from foreign key constraints", {
```

Comment:
This seems heavier and less clear to me than something that is just pushing data around.  Why are mocks needed here?

Reply:
To use the `ForeignKeyConstraints` object we'd have to setup a test db with the constraints which cover all the edge cases that the `constraint_table data` has in at the moment. That seemed more work than just mocking out the functions which use the db connection.

I could avoid this by having the `get_network_table` function on the object call a separate function which takes the table of constraints and the data and unit testing that instead.